datadog_checks_base/datadog_checks/base/log.py

```diff
@@ -23,16 +23,35 @@ def trace(self, msg, *args, **kwargs):
             self._log(TRACE_LEVEL, msg, args, **kwargs)
 
 
+class CheckLoggingAdapter(logging.LoggerAdapter):
+    def __init__(self, logger, check):
+        super(CheckLoggingAdapter, self).__init__(logger, {})
+        self.check = check
+        self.check_id = self.check.check_id
+
+    def process(self, msg, kwargs):
+        if not self.check_id:
+            self.check_id = self.check.check_id
+            self.extra['_check_id'] = self.check_id or 'unknown'
+
+        kwargs['extra'] = self.extra
+        return msg, kwargs
```

Comment:
Do we need to cache `self.check_id` ? Would this work equally ? =>

```
    def process(self, msg, kwargs):
        self.extra['_check_id'] = self.check.check_id or 'unknown'
        kwargs['extra'] = self.extra
        return msg, kwargs
```

Reply:
For performance, yes